standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/HiveMetaStoreChecker.java

```diff
@@ -240,40 +243,27 @@ void checkTable(String catName, String dbName, String tableName,
     }
 
     PartitionIterable parts;
-    boolean findUnknownPartitions = true;
 
     if (isPartitioned(table)) {
-      if (partitions == null || partitions.isEmpty()) {
+      if (filterExp != null) {
+        List<Partition> results = new ArrayList<>();
+        getPartitionListByFilterExp(getMsc(), table, filterExp,
```

Comment:
I wonder if there is a way to retain `filterExp` in a more natural way....it will be kryo-encoded almost all the time...but seems like the metastore interface method was designed to accept kryo stuff...

Reply:
I though this initially but it becomes difficult to serialize filter exp once we move out of ql code, Hence did the kyro stuff in the initial stage

Reply:
yeah I understand...sometimes we have to cook from what we have :D

![gif](https://media.giphy.com/media/LVBH3rg1BUROw/giphy.gif)